polygeist/lib/Dialect/Polygeist/Transforms/LICM.cpp

```diff
@@ -173,42 +173,54 @@ class LoopGuardBuilder {
 public:
   static std::unique_ptr<LoopGuardBuilder> create(LoopLikeOpInterface loop);
 
-  LoopGuardBuilder(LoopLikeOpInterface loop) : builder(loop){};
+  LoopGuardBuilder(LoopLikeOpInterface loop) : builder(loop), loop(loop){};
   LoopGuardBuilder(const LoopGuardBuilder &) = delete;
   LoopGuardBuilder(LoopGuardBuilder &&) = delete;
   void operator=(const LoopGuardBuilder &) = delete;
   void operator=(LoopGuardBuilder &&) = delete;
   virtual ~LoopGuardBuilder() = default;
 
-  virtual void guardLoop() const = 0;
+  void guardLoop() const;
 
 protected:
+  static Block *getThenBlock(RegionBranchOpInterface ifOp) {
```

Comment:
Can these return a reference rather than a pointer to `Block` ?

Reply:
done.